libraries/DataFlash/LogStructure.h

```diff
@@ -34,6 +36,91 @@ struct PACKED log_Format {
     char labels[64];
 };
 
+struct PACKED log_Unit {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    char unit[64]; // you know, this might be overkill...
+};
+
+struct PACKED log_Format_Multiplier {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    double multiplier;
+};
+
+struct PACKED log_Format_Units {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t format_type;
+    char units[16];
+    char multipliers[16];
+};
+
+struct UnitStructure {
+    const char ID;
+    const char unit[64];
+};
+
+struct MultiplierStructure {
+    const char ID;
+    const double multiplier;
+};
+
+// all units here should be base units
+// This does mean battery capacity is here as "amps*seconds"
+const struct UnitStructure log_Units[] = {
+    { '-', "" },    // no units e.g. Pi, or a string
+    { '?', "UNKNOWN" },    // Units which haven't been worked out yet....
+    { 'A', "amps" },
+    { 'd', "degrees" }, // of the angular variety, -180 to 180
+    { 'D', "deglatitude" },
+    { 'b', "bytes" },
+    { 'k', "degrees/second" },
+    { 'e', "degrees/second/second" },
+    { 'E', "radians/second" },
+    { 'G', "gauss" },
+    { 'h', "degreesheading" }, // 0.? to 359.?
+    { 'i', "amps*seconds" },
+    { 'm', "metres" },
+    { 'n', "metres/second" },
+    // { 'N', "Newtons" },
+    { 'o', "metres/second/second" },
```

Comment:
meters per second per second

Reply:
Ah, no, this isn't English (or French.... ;-) )

This is supposed to be trivially machine-parsable so the GCS can convert into a unit the user is happy with, presumably with something arcane and nonsensical.

Reply:
This is to match the string in https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/param_metadata/param.py#L33
This should help the GCS software developers by using unified string sets, although is it machine parsed. And because it is machine parsed ... less different strings, less bugs, less fuss maintaining it.